storages/replicator.go

```diff
@@ -73,6 +78,29 @@ func (rc *ReplicationClient) Do(request *Request) <-chan BackendResponse {
 	}()
 	return responsesChan
 }
+func replicateRequest(ctx context.Context, request *Request) (*http.Request, error) {
+	replicatedRequest, err := http.NewRequest(request.Method, request.URL.String(), request.Body)
+	if resetter, ok := replicatedRequest.Body.(types.Resetter); ok {
+		replicatedRequest.Body = resetter.Reset()
+	}
+	if err != nil {
+		return nil, err
+	}
+	replicatedRequest.Header = http.Header{}
+	for headerName, headerValues := range request.Header {
+		for idx := range headerValues {
+			replicatedRequest.Header.Add(headerName, headerValues[idx])
+		}
+	}
+	replicatedRequest.Host = request.Host
+	replicatedRequest.ContentLength = request.ContentLength
+	replicatedRequest.Form = request.Form
+	replicatedRequest.PostForm = request.PostForm
+	replicatedRequest.GetBody = request.GetBody
+	replicatedRequest.Trailer = request.Trailer
+	replicatedRequest.TransferEncoding = request.TransferEncoding
```

Comment:
there is shortcut for that block: 
* replicatedRequest = *request
And it looks like code duplication, see sharding/shards_ring.go#L81:112

Reply:
The shortcut would overwrite the field, including body and etc